database/views.py

```diff
@@ -116,3 +122,174 @@ def get_success_url(self):
     def form_valid(self, form):
         form.save()
         return super().form_valid(form)
+
+
+class DynamicModelSetCreateView(DynamicModelMixin, FormView):
+    template_name = "crud/generic_crud_view.html"
+    context_object_name = 'all_objects'
+
+    def get_form_class(self):
+        return get_dynamic_model_choice_set_form(self.dynamic_model)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({
+            'all_objects': ModelSet.objects.filter(
+                setmembership__content_type__model=self.dynamic_model.__name__.lower()
+            ).distinct(),
+            'dynamic_model_name': self.dynamic_model_name,
+            'url_name': resolve(self.request.path_info).url_name
+        })
+
+        return context
+
+    def get_success_url(self):
+        return self.request.path
+
+    def form_valid(self, form):
+        SetMembership.objects.bulk_create([
+            SetMembership(
+                set=form.cleaned_data['set'],
+                member_object=choice
+            )
+            for choice in form.cleaned_data['choices']
+        ])
+        return super().form_valid(form)
+
+
+class DynamicModelSetUpdateView(DynamicModelMixin, FormView):
+    template_name = "crud/generic_update_view.html"
+    context_object_name = 'all_objects'
+    object = None
+
+    def get_form_class(self):
+        return get_dynamic_model_choice_set_form(self.dynamic_model)
+
+    def get_initial(self):
+        self.object = get_object_or_404(ModelSet, pk=self.kwargs.get('id'))
+        return {
+            'set': self.object,
+            'choices': self.dynamic_model.objects.filter(sets__set=self.object)
+        }
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({
+            'remove_set': True,
+        })
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['object'] = get_object_or_404(ModelSet, pk=self.kwargs.get('id'))
+        return context
+
+    def get_success_url(self):
+        return reverse('database:set_crud', kwargs={'model': self.dynamic_model_name})
+
+    def form_valid(self, form):
+        SetMembership.objects.filter(
+            set=form.cleaned_data['set'], content_type__model=self.dynamic_model.__name__.lower()
+        ).delete()
+        SetMembership.objects.bulk_create([
+            SetMembership(
+                set=form.cleaned_data['set'],
+                member_object=choice
+            )
+            for choice in form.cleaned_data['choices']
+        ])
+        return super().form_valid(form)
+
+
+class DynamicModelSetInspectView(DynamicModelMixin, TemplateView):
```

Comment:
These probably need some comments explaining how they work: its hard to parse what the code does by just reading it